src/main/php/lang/ast/syntax/PHP.class.php

```diff
@@ -1090,16 +1090,42 @@ public function __construct() {
     });
   }
 
-  private function type($parse, $optional= true) {
-    $t= [];
-    do {
-      $t[]= $this->type0($parse, $optional);
-      if ('|' === $parse->token->value) {
+  public function type($parse, $optional= true) {
+    $t= $this->type0($parse, $optional);
+
+    if ('|' === $parse->token->value) {
+      $t= new IsUnion([$t]);
+      do {
         $parse->forward();
-        continue;
+        $t->components[]= $this->type0($parse, false);
+      } while ('|' === $parse->token->value);
+    } else if ('&' === $parse->token->value) {
```

Comment:
The RFC states:

> This means it would not be possible to mix intersection and union types together such as A&B|C, this is left as a future scope.

This is why it's OK to check for `|` and for `&` but not to check for the respective other after parsing